app/src/main/java/ch/epfl/sdp/location/LocationService.java

```diff
@@ -223,16 +313,34 @@ public int onStartCommand(Intent intent, int flags, int startId) {
     @Override
     public void update(Observable o, Object arg) {
         // Store updates to Carrier
-        locallyStoreCarrier();
-        AsyncTask.execute(() -> remotelyStoreCarrierStatus(analyst.getCarrier().getInfectionStatus()));
+        AsyncTask.execute(() -> {
```

Comment:
Are you sure that you don't interact with UI components in theses functions ?

You have to declare a subclass as explained here if that's the case or you may run into some nasty concurrency issues: https://developer.android.com/reference/android/os/AsyncTask

Reply:
I'm quite sure

Reply:
I'm quite convinced then